Timeouts waiting to get a http connection

This is following the change to use dropwizard configured jersey clients

```
ERROR  [2023-05-12T06:53:12.836Z] [Stroom P2 #1] stroom.processor.impl.DataProcessorTaskFactory -  javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
java.lang.RuntimeException: javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
        at stroom.node.api.NodeCallUtil.handleExceptionsOnNodeCall(NodeCallUtil.java:58)
        at stroom.node.impl.NodeServiceImpl.remoteRestResult(NodeServiceImpl.java:198)
        at stroom.node.api.NodeService.remoteRestResult(NodeService.java:69)
        at stroom.processor.impl.ProcessorTaskResourceImpl.assignTasks(ProcessorTaskResourceImpl.java:125)
        at stroom.processor.impl.DataProcessorTaskFactory.fetch(DataProcessorTaskFactory.java:74)
        at stroom.job.impl.DistributedTaskFetcher.doFetch(DistributedTaskFetcher.java:209)
        at stroom.job.impl.DistributedTaskFetcher.lambda$fetch$0(DistributedTaskFetcher.java:131)
        at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$2(TaskContextFactoryImpl.java:272)
        at stroom.util.logging.LocationAwareLambdaLogger.logDurationIfDebugEnabled(LocationAwareLambdaLogger.java:149)
        at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$4(TaskContextFactoryImpl.java:272)
        at stroom.util.pipeline.scope.PipelineScopeRunnable.scopeResult(PipelineScopeRunnable.java:38)
        at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$5(TaskContextFactoryImpl.java:269)
        at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$7(TaskContextFactoryImpl.java:283)
        at stroom.security.impl.SecurityContextImpl.asUserResult(SecurityContextImpl.java:261)
        at jdk.internal.reflect.GeneratedMethodAccessor163.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at com.google.inject.internal.DelegatingInvocationHandler.invoke(DelegatingInvocationHandler.java:50)
        at jdk.proxy2/jdk.proxy2.$Proxy109.asUserResult(Unknown Source)
        at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$11(TaskContextFactoryImpl.java:279)
        at stroom.job.impl.DistributedTaskFetcher.lambda$fetch$1(DistributedTaskFetcher.java:136)
        at stroom.security.impl.SecurityContextImpl.asUser(SecurityContextImpl.java:279)
        at stroom.security.impl.SecurityContextImpl.asProcessingUser(SecurityContextImpl.java:300)
        at stroom.job.impl.DistributedTaskFetcher.fetch(DistributedTaskFetcher.java:124)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
        at io.dropwizard.client.DropwizardApacheConnector.apply(DropwizardApacheConnector.java:105)
        at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:297)
        at org.glassfish.jersey.client.JerseyInvocation.lambda$invoke$0(JerseyInvocation.java:662)
        at org.glassfish.jersey.client.JerseyInvocation.call(JerseyInvocation.java:697)
        at org.glassfish.jersey.client.JerseyInvocation.lambda$runInScope$3(JerseyInvocation.java:691)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:205)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:390)
        at org.glassfish.jersey.client.JerseyInvocation.runInScope(JerseyInvocation.java:691)
        at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:661)
        at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:439)
        at org.glassfish.jersey.client.JerseyInvocation$Builder.post(JerseyInvocation.java:345)
        at stroom.processor.impl.ProcessorTaskResourceImpl.lambda$assignTasks$2(ProcessorTaskResourceImpl.java:136)
        at stroom.node.impl.NodeServiceImpl.remoteRestResult(NodeServiceImpl.java:188)
        ... 25 common frames omitted



ERROR  [2023-05-12T06:52:33.723Z] [Stroom P2 #1285] stroom.cluster.lock.impl.db.ClusterLockHandler -  Error connecting to master node 'xxxxxxxxxxxxxxxxxxxx' - javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
java.lang.RuntimeException: javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
        at stroom.node.api.NodeCallUtil.handleExceptionsOnNodeCall(NodeCallUtil.java:58)
        at stroom.cluster.lock.impl.db.ClusterLockResourceImpl.executeRemotely(ClusterLockResourceImpl.java:98)
        at stroom.cluster.lock.impl.db.ClusterLockResourceImpl.tryLock(ClusterLockResourceImpl.java:60)
        at stroom.cluster.lock.impl.db.ClusterLockHandler.lambda$tryLock$0(ClusterLockHandler.java:50)
        at stroom.cluster.lock.impl.db.ClusterLockHandler.lambda$call$3(ClusterLockHandler.java:65)
        at stroom.security.impl.SecurityContextImpl.secureResult(SecurityContextImpl.java:473)
        at stroom.cluster.lock.impl.db.ClusterLockHandler.call(ClusterLockHandler.java:65)
        at stroom.cluster.lock.impl.db.ClusterLockHandler.tryLock(ClusterLockHandler.java:50)
        at stroom.cluster.lock.impl.db.ClusterLockServiceImpl.tryLock(ClusterLockServiceImpl.java:81)
        at stroom.processor.impl.ProcessorTaskCreatorImpl.exec(ProcessorTaskCreatorImpl.java:144)
        at stroom.util.RunnableWrapper.run(RunnableWrapper.java:19)
        at stroom.job.impl.ScheduledJobFunction.run(ScheduledJobFunction.java:27)
        at stroom.job.impl.ScheduledTaskExecutor$JobNodeTrackedFunction.run(ScheduledTaskExecutor.java:270)
        at stroom.job.impl.ScheduledTaskExecutor.lambda$execute$2(ScheduledTaskExecutor.java:149)
        at stroom.task.impl.TaskContextFactoryImpl.lambda$createFromConsumer$0(TaskContextFactoryImpl.java:195)
        at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$2(TaskContextFactoryImpl.java:272)
        at stroom.util.logging.LocationAwareLambdaLogger.logDurationIfDebugEnabled(LocationAwareLambdaLogger.java:149)
        at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$4(TaskContextFactoryImpl.java:272)
        at stroom.util.pipeline.scope.PipelineScopeRunnable.scopeResult(PipelineScopeRunnable.java:38)
        at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$5(TaskContextFactoryImpl.java:269)
        at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$7(TaskContextFactoryImpl.java:283)
        at stroom.security.impl.SecurityContextImpl.asUserResult(SecurityContextImpl.java:261)
        at jdk.internal.reflect.GeneratedMethodAccessor163.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at com.google.inject.internal.DelegatingInvocationHandler.invoke(DelegatingInvocationHandler.java:50)
        at jdk.proxy2/jdk.proxy2.$Proxy109.asUserResult(Unknown Source)
        at stroom.task.impl.TaskContextFactoryImpl.lambda$wrap$11(TaskContextFactoryImpl.java:279)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
        at io.dropwizard.client.DropwizardApacheConnector.apply(DropwizardApacheConnector.java:105)
        at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:297)
        at org.glassfish.jersey.client.JerseyInvocation.lambda$invoke$0(JerseyInvocation.java:662)
        at org.glassfish.jersey.client.JerseyInvocation.call(JerseyInvocation.java:697)
        at org.glassfish.jersey.client.JerseyInvocation.lambda$runInScope$3(JerseyInvocation.java:691)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:205)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:390)
        at org.glassfish.jersey.client.JerseyInvocation.runInScope(JerseyInvocation.java:691)
        at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:661)
        at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:439)
        at org.glassfish.jersey.client.JerseyInvocation$Builder.put(JerseyInvocation.java:328)
        at stroom.cluster.lock.impl.db.ClusterLockResourceImpl.executeRemotely(ClusterLockResourceImpl.java:92)
        ... 30 common frames omitted
Caused by: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
        at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.leaseConnection(PoolingHttpClientConnectionManager.java:316)
        at org.apache.http.impl.conn.PoolingHttpClientConnectionManager$1.get(PoolingHttpClientConnectionManager.java:282)
        at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:190)
        at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
        at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
        at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
        at io.dropwizard.client.DropwizardApacheConnector.apply(DropwizardApacheConnector.java:87)
        ... 43 common frames omitted
		INFO   [2023-05-12T06:52:33.723Z] [Stroom P2 #1285] stroom.cluster.lock.impl.db.ClusterLockServiceImpl -  Skipped process as did not get lock ProcessorTaskCreator in 30s
WARN   [2023-05-12T06:52:45.920Z] [Stroom P2 #1298] stroom.cluster.task.impl.ClusterTaskTerminatorImpl -  javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
WARN   [2023-05-12T06:52:45.921Z] [Stroom P2 #1287] stroom.cluster.task.impl.ClusterTaskTerminatorImpl -  javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
WARN   [2023-05-12T06:52:45.921Z] [Stroom P2 #1295] stroom.cluster.task.impl.ClusterTaskTerminatorImpl -  javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
WARN   [2023-05-12T06:52:45.922Z] [Stroom P2 #1272] stroom.cluster.task.impl.ClusterTaskTerminatorImpl -  javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
WARN   [2023-05-12T06:52:45.924Z] [Stroom P2 #1244] stroom.cluster.impl.ClusterNodeManagerImpl -  discover() - unable to contact xxxxxxx - javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
WARN   [2023-05-12T06:52:45.924Z] [Stroom P2 #1288] stroom.cluster.impl.ClusterNodeManagerImpl -  discover() - unable to contact xxxxxxx - javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
INFO   [2023-05-12T06:52:45.924Z] [Stroom P2 #1288] stroom.cluster.impl.ClusterNodeManagerImpl -  Master node has changed from xxxxxxxx to xxxxxxxxxxxxxxxxxxxxxxxxxx
WARN   [2023-05-12T06:52:46.723Z] [Stroom P2 #1271] stroom.cluster.task.impl.ClusterTaskTerminatorImpl -  javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
WARN   [2023-05-12T06:52:46.723Z] [Stroom P2 #1289] stroom.cluster.task.impl.ClusterTaskTerminatorImpl -  javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
WARN   [2023-05-12T06:52:46.723Z] [Stroom P2 #1282] stroom.cluster.task.impl.ClusterTaskTerminatorImpl -  javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
WARN   [2023-05-12T06:52:46.723Z] [Stroom P2 #1294] stroom.cluster.task.impl.ClusterTaskTerminatorImpl -  javax.ws.rs.ProcessingException: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
```

--- a/stroom-cache/stroom-cache-impl/src/main/java/stroom/cache/impl/CacheResourceImpl.java
+++ b/stroom-cache/stroom-cache-impl/src/main/java/stroom/cache/impl/CacheResourceImpl.java
@@ -89,13 +89,14 @@ class CacheResourceImpl implements CacheResource {
             try {
                 WebTarget webTarget = webTargetFactory.get().create(url);
                 webTarget = UriBuilderUtil.addParam(webTarget, "nodeName", nodeName);
-                final Response response = webTarget
+                try (Response response = webTarget
                         .request(MediaType.APPLICATION_JSON)
-                        .get();
-                if (response.getStatus() != 200) {
-                    throw new WebApplicationException(response);
+                        .get()) {
+                    if (response.getStatus() != 200) {
+                        throw new WebApplicationException(response);
+                    }
+                    result = response.readEntity(CacheNamesResponse.class);
                 }
-                result = response.readEntity(CacheNamesResponse.class);
                 if (result == null) {
                     throw new RuntimeException("Unable to contact node \"" + nodeName + "\" at URL: " + url);
                 }
@@ -123,14 +124,15 @@ class CacheResourceImpl implements CacheResource {
                 WebTarget webTarget = webTargetFactory.get().create(url);
                 webTarget = UriBuilderUtil.addParam(webTarget, "cacheName", cacheName);
                 webTarget = UriBuilderUtil.addParam(webTarget, "nodeName", nodeName);
-                final Response response = webTarget
-                        .request(MediaType.APPLICATION_JSON)
-                        .get();
-                if (response.getStatus() != 200) {
-                    throw new WebApplicationException(response);
-                }
                 CacheInfoResponse result;
-                result = response.readEntity(CacheInfoResponse.class);
+                try (Response response = webTarget
+                        .request(MediaType.APPLICATION_JSON)
+                        .get()) {
+                    if (response.getStatus() != 200) {
+                        throw new WebApplicationException(response);
+                    }
+                    result = response.readEntity(CacheInfoResponse.class);
+                }
                 if (result == null) {
                     throw new RuntimeException("Unable to contact node \"" + nodeName + "\" at URL: " + url);
                 }
@@ -232,13 +234,14 @@ class CacheResourceImpl implements CacheResource {
                 WebTarget webTarget = webTargetFactory.get().create(url);
                 webTarget = UriBuilderUtil.addParam(webTarget, "cacheName", cacheName);
                 webTarget = UriBuilderUtil.addParam(webTarget, "nodeName", nodeName);
-                final Response response = webTarget
+                try (Response response = webTarget
                         .request(MediaType.APPLICATION_JSON)
-                        .delete();
-                if (response.getStatus() != 200) {
-                    throw new WebApplicationException(response);
+                        .delete()) {
+                    if (response.getStatus() != 200) {
+                        throw new WebApplicationException(response);
+                    }
+                    result = response.readEntity(Long.class);
                 }
-                result = response.readEntity(Long.class);
             } catch (Throwable e) {
                 throw NodeCallUtil.handleExceptionsOnNodeCall(nodeName, url, e);
             }

--- a/stroom-cluster/stroom-cluster-lock-impl-db/src/main/java/stroom/cluster/lock/impl/db/ClusterLockResourceImpl.java
+++ b/stroom-cluster/stroom-cluster-lock-impl-db/src/main/java/stroom/cluster/lock/impl/db/ClusterLockResourceImpl.java
@@ -86,14 +86,15 @@ class ClusterLockResourceImpl implements ClusterLockResource {
                 nodeName);
 
         try {
-            final Response response = webTargetFactory.get()
+            try (Response response = webTargetFactory.get()
                     .create(url)
                     .request(MediaType.APPLICATION_JSON)
-                    .put(Entity.json(key));
-            if (response.getStatus() != 200) {
-                throw new WebApplicationException(response);
+                    .put(Entity.json(key))) {
+                if (response.getStatus() != 200) {
+                    throw new WebApplicationException(response);
+                }
+                return response.readEntity(Boolean.class);
             }
-            return response.readEntity(Boolean.class);
         } catch (Throwable e) {
             throw NodeCallUtil.handleExceptionsOnNodeCall(nodeName, url, e);
         }

--- a/stroom-core/src/main/java/stroom/core/entity/event/EntityEventResourceImpl.java
+++ b/stroom-core/src/main/java/stroom/core/entity/event/EntityEventResourceImpl.java
@@ -67,15 +67,17 @@ class EntityEventResourceImpl implements EntityEventResource {
                     nodeName);
 
             try {
-                final Response response = webTargetFactoryProvider
+                final Boolean success;
+                try (Response response = webTargetFactoryProvider
                         .get()
                         .create(url)
                         .request(MediaType.APPLICATION_JSON)
-                        .put(Entity.json(entityEvent));
-                if (response.getStatus() != 200) {
-                    throw new WebApplicationException(response);
+                        .put(Entity.json(entityEvent))) {
+                    if (response.getStatus() != 200) {
+                        throw new WebApplicationException(response);
+                    }
+                    success = response.readEntity(Boolean.class);
                 }
-                final Boolean success = response.readEntity(Boolean.class);
                 Objects.requireNonNull(success, "Null success");
                 return success;
             } catch (Throwable e) {

--- a/stroom-node/stroom-node-impl/src/main/java/stroom/node/impl/NodeServiceImpl.java
+++ b/stroom-node/stroom-node-impl/src/main/java/stroom/node/impl/NodeServiceImpl.java
@@ -185,13 +185,14 @@ public class NodeServiceImpl implements NodeService {
             try {
                 final Builder builder = createBuilder(queryParams, url);
 
-                final Response response = responseBuilderFunc.apply(builder);
+                try (Response response = responseBuilderFunc.apply(builder)) {
 
-                LOGGER.debug(() -> "Response status " + response.getStatus());
-                if (response.getStatus() != Status.OK.getStatusCode()) {
-                    throw new WebApplicationException(response);
+                    LOGGER.debug(() -> "Response status " + response.getStatus());
+                    if (response.getStatus() != Status.OK.getStatusCode()) {
+                        throw new WebApplicationException(response);
+                    }
+                    resp = responseMapper.apply(response);
                 }
-                resp = responseMapper.apply(response);
 
                 Objects.requireNonNull(resp, "Null response calling url " + url);
             } catch (final Throwable e) {
@@ -225,11 +226,12 @@ public class NodeServiceImpl implements NodeService {
             try {
                 final Builder builder = createBuilder(queryParams, url);
 
-                final Response response = responseBuilderFunc.apply(builder);
+                try (Response response = responseBuilderFunc.apply(builder)) {
 
-                LOGGER.debug(() -> "Response status " + response.getStatus());
-                if (response.getStatus() != Status.OK.getStatusCode()) {
-                    throw new WebApplicationException(response);
+                    LOGGER.debug(() -> "Response status " + response.getStatus());
+                    if (response.getStatus() != Status.OK.getStatusCode()) {
+                        throw new WebApplicationException(response);
+                    }
                 }
             } catch (final Throwable e) {
                 throw NodeCallUtil.handleExceptionsOnNodeCall(nodeName, url, e);

--- a/stroom-pipeline/src/main/java/stroom/pipeline/filter/HttpPostFilter.java
+++ b/stroom-pipeline/src/main/java/stroom/pipeline/filter/HttpPostFilter.java
@@ -78,17 +78,18 @@ public class HttpPostFilter extends AbstractSamplingFilter {
         } else {
             try {
                 final Client client = jerseyClientFactory.getNamedClient(JerseyClientName.HTTP_POST_FILTER);
-                final Response response = client
+                try (Response response = client
                         .target(receivingApiUrl)
                         .request()
                         .accept(MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON)
-                        .post(Entity.xml(xml));
-                if (response.getStatus() != Response.Status.ACCEPTED.ordinal()) {
-                    final String errorMessage = String.format("POST was not accepted by the API: %s", response);
-                    errorReceiverProxy.log(Severity.ERROR, null, null, errorMessage, null);
-                    LOGGER.error(errorMessage);
-                } else {
-                    LOGGER.info("POSTed document to API {}", receivingApiUrl);
+                        .post(Entity.xml(xml))) {
+                    if (response.getStatus() != Response.Status.ACCEPTED.ordinal()) {
+                        final String errorMessage = String.format("POST was not accepted by the API: %s", response);
+                        errorReceiverProxy.log(Severity.ERROR, null, null, errorMessage, null);
+                        LOGGER.error(errorMessage);
+                    } else {
+                        LOGGER.info("POSTed document to API {}", receivingApiUrl);
+                    }
                 }
             } catch (final RuntimeException e) {
                 final String errorMessage = String.format("Unable to POST document to API: %s", e);

--- a/stroom-proxy/stroom-proxy-app/src/main/java/stroom/proxy/app/ContentSyncService.java
+++ b/stroom-proxy/stroom-proxy-app/src/main/java/stroom/proxy/app/ContentSyncService.java
@@ -112,14 +112,15 @@ public class ContentSyncService implements Managed, HasHealthCheck {
                     try {
                         if (url != null) {
                             LOGGER.info("Syncing content from '" + url + "'");
-                            final Response response = createClient(url, "/list").get();
-                            if (response.getStatusInfo().getStatusCode() != Status.OK.getStatusCode()) {
-                                LOGGER.error(response.getStatusInfo().getReasonPhrase());
-                            } else {
-                                final Set<DocRef> docRefs = response.readEntity(new GenericType<Set<DocRef>>() {
-                                });
-                                docRefs.forEach(docRef -> importDocument(url, docRef, importHandler));
-                                LOGGER.info("Synced {} documents", docRefs.size());
+                            try (Response response = createClient(url, "/list").get()) {
+                                if (response.getStatusInfo().getStatusCode() != Status.OK.getStatusCode()) {
+                                    LOGGER.error(response.getStatusInfo().getReasonPhrase());
+                                } else {
+                                    final Set<DocRef> docRefs = response.readEntity(new GenericType<Set<DocRef>>() {
+                                    });
+                                    docRefs.forEach(docRef -> importDocument(url, docRef, importHandler));
+                                    LOGGER.info("Synced {} documents", docRefs.size());
+                                }
                             }
                         }
                     } catch (Exception e) {
@@ -134,19 +135,20 @@ public class ContentSyncService implements Managed, HasHealthCheck {
                                 final DocRef docRef,
                                 final ImportExportActionHandler importExportActionHandler) {
         LOGGER.info("Fetching " + docRef.getType() + " " + docRef.getName() + " " + docRef.getUuid());
-        final Response response = createClient(url, "/export").post(Entity.json(docRef));
-        if (response.getStatusInfo().getStatusCode() != Status.OK.getStatusCode()) {
-            LOGGER.error(response.getStatusInfo().getReasonPhrase());
-        } else {
-            final DocumentData documentData = response.readEntity(DocumentData.class);
-            final ImportState importState = new ImportState(
-                    documentData.getDocRef(),
-                    documentData.getDocRef().getName());
-            importExportActionHandler.importDocument(
-                    documentData.getDocRef(),
-                    documentData.getDataMap(),
-                    importState,
-                    ImportSettings.auto());
+        try (Response response = createClient(url, "/export").post(Entity.json(docRef))) {
+            if (response.getStatusInfo().getStatusCode() != Status.OK.getStatusCode()) {
+                LOGGER.error(response.getStatusInfo().getReasonPhrase());
+            } else {
+                final DocumentData documentData = response.readEntity(DocumentData.class);
+                final ImportState importState = new ImportState(
+                        documentData.getDocRef(),
+                        documentData.getDocRef().getName());
+                importExportActionHandler.importDocument(
+                        documentData.getDocRef(),
+                        documentData.getDataMap(),
+                        importState,
+                        ImportSettings.auto());
+            }
         }
     }
 

--- a/stroom-proxy/stroom-proxy-app/src/test/java/stroom/proxy/app/PostDataHelper.java
+++ b/stroom-proxy/stroom-proxy-app/src/test/java/stroom/proxy/app/PostDataHelper.java
@@ -85,10 +85,12 @@ public class PostDataHelper {
                 extraHeaders.forEach(builder::header);
             }
             LOGGER.info("Sending POST request to {}", url);
-            final Response response = builder.post(Entity.text(data));
-            postToProxyCount.increment();
-            status = response.getStatus();
-            final String responseText = response.readEntity(String.class);
+            final String responseText;
+            try (Response response = builder.post(Entity.text(data))) {
+                postToProxyCount.increment();
+                status = response.getStatus();
+                responseText = response.readEntity(String.class);
+            }
             LOGGER.info("datafeed response ({}):\n{}", status, responseText);
 
         } catch (final Exception e) {
@@ -126,11 +128,13 @@ public class PostDataHelper {
                 }
             }
 
-            final Response response = builder.post(
-                    Entity.entity(outputStream.toByteArray(), MediaType.APPLICATION_JSON_TYPE));
-            postToProxyCount.increment();
-            status = response.getStatus();
-            final String responseText = response.readEntity(String.class);
+            final String responseText;
+            try (Response response = builder.post(
+                    Entity.entity(outputStream.toByteArray(), MediaType.APPLICATION_JSON_TYPE))) {
+                postToProxyCount.increment();
+                status = response.getStatus();
+                responseText = response.readEntity(String.class);
+            }
             LOGGER.info("datafeed response ({}):\n{}", status, responseText);
 
         } catch (final Exception e) {

--- a/stroom-search/stroom-search-impl/src/main/java/stroom/search/impl/RemoteNodeSearch.java
+++ b/stroom-search/stroom-search-impl/src/main/java/stroom/search/impl/RemoteNodeSearch.java
@@ -117,17 +117,18 @@ public class RemoteNodeSearch implements NodeSearch {
                 RemoteSearchResource.START_PATH_PART);
 
         try {
-            final Response response = webTargetFactory
+            try (Response response = webTargetFactory
                     .create(url)
                     .request(MediaType.APPLICATION_JSON)
-                    .post(Entity.json(clusterSearchTask));
-            if (response.getStatus() == Status.NOT_FOUND.getStatusCode()) {
-                throw new NotFoundException(response);
-            } else if (response.getStatus() != Status.OK.getStatusCode()) {
-                throw new WebApplicationException(response);
-            }
+                    .post(Entity.json(clusterSearchTask))) {
+                if (response.getStatus() == Status.NOT_FOUND.getStatusCode()) {
+                    throw new NotFoundException(response);
+                } else if (response.getStatus() != Status.OK.getStatusCode()) {
+                    throw new WebApplicationException(response);
+                }
 
-            return response.readEntity(Boolean.class);
+                return response.readEntity(Boolean.class);
+            }
         } catch (Throwable e) {
             LOGGER.debug(e::getMessage, e);
             throw NodeCallUtil.handleExceptionsOnNodeCall(nodeName, url, e);
@@ -167,16 +168,17 @@ public class RemoteNodeSearch implements NodeSearch {
             WebTarget webTarget = webTargetFactory.create(url);
             webTarget = UriBuilderUtil.addParam(webTarget, "queryKey", queryKey);
 
-            final Response response = webTarget
+            try (Response response = webTarget
                     .request(MediaType.APPLICATION_JSON)
-                    .get();
-            if (response.getStatus() == Status.NOT_FOUND.getStatusCode()) {
-                throw new NotFoundException(response);
-            } else if (response.getStatus() != Status.OK.getStatusCode()) {
-                throw new WebApplicationException(response);
-            }
+                    .get()) {
+                if (response.getStatus() == Status.NOT_FOUND.getStatusCode()) {
+                    throw new NotFoundException(response);
+                } else if (response.getStatus() != Status.OK.getStatusCode()) {
+                    throw new WebApplicationException(response);
+                }
 
-            return response.readEntity(Boolean.class);
+                return response.readEntity(Boolean.class);
+            }
         } catch (Throwable e) {
             LOGGER.debug(e::getMessage, e);
             throw NodeCallUtil.handleExceptionsOnNodeCall(nodeName, url, e);

--- a/stroom-security/stroom-security-impl/src/main/java/stroom/security/impl/SessionListListener.java
+++ b/stroom-security/stroom-security-impl/src/main/java/stroom/security/impl/SessionListListener.java
@@ -128,15 +128,16 @@ class SessionListListener implements HttpSessionListener, SessionListService {
                 LOGGER.debug("Sending request to {} for node {}", url, nodeName);
                 WebTarget webTarget = webTargetFactory.create(url);
                 webTarget = UriBuilderUtil.addParam(webTarget, SessionResource.NODE_NAME_PARAM, nodeName);
-                final Response response = webTarget
+                try (Response response = webTarget
                         .request(MediaType.APPLICATION_JSON)
-                        .get();
+                        .get()) {
 
-                if (response.getStatus() != 200) {
-                    throw new WebApplicationException(response);
+                    if (response.getStatus() != 200) {
+                        throw new WebApplicationException(response);
+                    }
+
+                    sessionList = response.readEntity(SessionListResponse.class);
                 }
-
-                sessionList = response.readEntity(SessionListResponse.class);
             } catch (Throwable e) {
                 throw NodeCallUtil.handleExceptionsOnNodeCall(nodeName, url, e);
             }

--- a/unreleased_changes/20230512_143219_864__3449.md
+++ b/unreleased_changes/20230512_143219_864__3449.md
@@ -1,0 +1,24 @@
+* Issue **#3449** : Ensure jersey client responses are closed properly.
+
+
+```sh
+# ********************************************************************************
+# Issue title: Timeouts waiting to get a http connection
+# Issue link:  https://github.com/gchq/stroom/issues/3449
+# ********************************************************************************
+
+# ONLY the top line will be included as a change entry in the CHANGELOG.
+# The entry should be in GitHub flavour markdown and should be written on a SINGLE
+# line with no hard breaks. You can have multiple change files for a single GitHub issue.
+# The  entry should be written in the imperative mood, i.e. 'Fix nasty bug' rather than
+# 'Fixed nasty bug'.
+#
+# Examples of acceptable entries are:
+#
+#
+# * Issue **123** : Fix bug with an associated GitHub issue in this repository
+#
+# * Issue **namespace/other-repo#456** : Fix bug with an associated GitHub issue in another repository
+#
+# * Fix bug with no associated GitHub issue.
+```

